pkg/controller/machine.go

```diff
@@ -362,7 +388,18 @@ func (c *controller) machineCreate(machine *v1alpha1.Machine, driver driver.Driv
 		c.updateMachineStatus(machine, lastOperation, currentStatus)
 		return err
 	}
-	actualProviderID, nodeName, err := driver.Create()
+	// Before actually creating the machine, we should once check and adopt if the virtual machine already exists.
+	VMList, err := driver.GetVMs("")
+	for providerID, machineName := range VMList {
+		if machineName == machine.Name {
+			klog.V(2).Infof("Adopted an existing VM %s for machine object %s.", providerID, machineName)
+			actualProviderID = providerID
+		}
+	}
+	if actualProviderID == "" {
+		actualProviderID, nodeName, err = driver.Create()
+	}
+
 	if err != nil {
```

Comment:
This now ends up checking `err` for the added call `VMList, err := driver.GetVMs("")` instead of `actualProviderID, nodeName, err = driver.Create()`.
Error check needs to be added separately for the added call.

Reply:
Good catch.

Reply:
Hm, intersting catch. So basically if `GetVMs` fails, it's better to err back than creating the new machine. I'll make a change here to handle the error, thanks.

Reply:
done